src/apps/activity_assignments/db/schemas.py

```diff
@@ -0,0 +1,15 @@
+from sqlalchemy import Column, ForeignKey
+
+from infrastructure.database import Base
+
+__all__ = ["ActivityAssigmentSchema"]
+
+
+class ActivityAssigmentSchema(Base):
+    __tablename__ = "activity_assignments"
+
+    activity_flow_id = Column(ForeignKey("flows.id", ondelete="RESTRICT"), nullable=True)
+    activity_id = Column(ForeignKey("activities.id", ondelete="RESTRICT"), nullable=True)
+    invitation_id = Column(ForeignKey("invitations.id", ondelete="RESTRICT"), nullable=True)
+    respondent_id = Column(ForeignKey("users.id", ondelete="RESTRICT"), nullable=True)
+    target_subject_id = Column(ForeignKey("subjects.id", ondelete="RESTRICT"), nullable=True)
```

Comment:
I've just identified two issues with the schema in its current design, and I'm going to propose a change to it that I _hope_ will be fairly straightforward for you to make.

#### (1) FE doesn't know what the value of `invitation_id` is
In the FE, we fetch the list of available respondents for activity assignment using the `/workspaces/{owner_id}/applets/{applet_id}/respondents` endpoint. However, that endpoint doesn't include `invitation_id` in the `invitation: { … }` part of the respondent details. In fact, `id` doesn't even seem to be returned by the `/invitation/*` endpoints themselves, so the FE simply has no way of knowing what that value is for pending respondents.

What the FE _does_ know is the `subject_id` of a participant, which the how we uniquely identify participants in all participant lists and dropdowns on the FE. The beauty of `subject_id` is that it's universally applicable to respondents, pending invitations, and target subjects. So I propose that we replace `invitation_id` and `respondent_id` props with, simply, `respondent_subject_id`, and make it `nullable=False`. This will make sending assignments to the API super simple from the FE's perspective, with no extra logic needed to be run based on invitation status.

#### (2) Allowing _nullable_ `target_subject_id` makes it possible to create duplicate self-assignments

I don't think `target_subject_id` should be allowed to be `nullable=True`. I was able to successfully create two self-assignment records in `activity_assignments` table, one where `target_subject_id` is `null`, and another one where `target_subject_id` is the subject ID of the respondent. I don't think it should be possible to have duplicate records that represent that same logical assignment (self-assignment).

---

So to summarize, I propose the schema be modified to:

```suggestion
    respondent_subject_id = Column(ForeignKey("users.id", ondelete="RESTRICT"), nullable=False)
    target_subject_id = Column(ForeignKey("subjects.id", ondelete="RESTRICT"), nullable=False)
```

This will make it straightforward for the Activity Assign screen to post assignments to the API, since this is the approximate data structure already in use on the FE:

```tsx
{
  activityId?: string;
  activityFlowId?: string;
  respondentSubjectId: string;
  targetSubjectId: string;
}
```

@rcmerlo happy to hear your input on this feedback. Do you foresee any difficulties in using my proposed revised schema?